### PR TITLE
Style for Filecoin.io community

### PR DESCRIPTION
**Pull request by IgorAlexeew**

### CODE REVIEW

File: app/communities-config.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import TelosLogo from './communities-configs/telos/images/telos-logo-dark.svg?inline';
 import KandaLogo from './communities-configs/kanda/images/kanda.jpg';
 
 import { TelosStyles } from './communities-configs/telos';
 import { OntStyles } from './communities-configs/ont';
 import { KandaStyles } from './communities-configs/kanda';
 import { BloggerStyles } from './communities-configs/blogger';
-
+import { FilecoinStyles } from './communities-configs/filecoin';
 const communitiesConfig = {
```
Comment (ruslanzharkov) [inline app/communities-config.js:11]: please add empty line after import

File: app/communities-configs/filecoin/images/filecoin-logo.svg [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/images/filecoin-logo.svg:1]: Did you have a change to optimize it via svg optimizers like this one?
https://jakearchibald.github.io/svgomg/
image
I see that we are able to reduce size of icon a little

File: app/communities-configs/filecoin/index.js [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+  extraBg: '#F3F3F3',
+  menuItemText: '#999999',
+  selectedMenuItemText: '#000000',
+};
+
+export const CustomSubHeaderConfig = {
+  design: 'filecoin_style',
+  styles: {
+    bg: {
+      header: colors.bg ?? '#FFFFFF',
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:31]: You have different style of colors: uppercase and lowercase `FFFFFF/ffffff`.
Also you can import some colors from `style-constants.js`. Also your unique colors can me moved to it's own file with all related colors

File: app/communities-configs/filecoin/index.js [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+        left: 50%;
+        transform: translateX(-50%) rotate(45deg);
+        transition: background-color .3s;
+        border-top: 1px solid #f2f2f2;
+        border-left: 1px solid #f2f2f2;
+        background: ${colors.bg};
+        z-index: 9999;
+      }
+
+      /*> div:last-child::after {
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:89]: should be removed if it's commented

File: app/communities-configs/filecoin/index.js [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+      a, span {
+        transition: all .3s ease-out;
+      }
+      a:hover, span:hover {
+        color: ${colors.selectedMenuItemText}!important;
+        opacity: 1!important;
+      }
+    `,
+    subitems: `
+      left: 50%!important;
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:54]: please use `css` from styled-components

File: app/communities-configs/filecoin/index.js [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+      top: 100%!important;
+      /*transform:translate(-50%, -5%)!important;*/
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:56]: remove this one

File: app/communities-configs/filecoin/index.js [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+      padding: 25px 30px !important;
+      border-radius: 3px!important;
+      min-width: 100%!important;
+      transform:translate(-50%, -5%)!important;
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:60]:
```suggestion
      transform: translate(-50%, -5%)!important;
```

File: app/communities-configs/filecoin/index.js [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+        opacity: 1!important;
+      }
+    `,
+    subitems: `
+      left: 50%!important;
+      top: 100%!important;
+      /*transform:translate(-50%, -5%)!important;*/
+      padding: 25px 30px !important;
+      border-radius: 3px!important;
+      min-width: 100%!important;
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:59]:
```suggestion
      min-width: 100% !important;
```

File: app/communities-configs/filecoin/index.js [hunk 9]
```diff
@@ -0,0 +1,318 @@
...
+    header: {
+      background: colors.bg ?? '#ffffff',
+    },
+    subHeader: `
+      border-bottom: 1px solid #CECECE;
+      a, span {
+        transition: all .3s ease-out;
+      }
+      a:hover, span:hover {
+        color: ${colors.selectedMenuItemText}!important;
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:49]:
```suggestion
        color: ${colors.selectedMenuItemText} !important;
```

File: app/communities-configs/filecoin/index.js [hunk 10]
```diff
@@ -0,0 +1,318 @@
...
+
+export const CustomSubHeaderConfig = {
+  design: 'filecoin_style',
+  styles: {
+    bg: {
+      header: colors.bg ?? '#FFFFFF',
+      dropdown: colors.bg ?? '#FFFFFF',
+    },
+    color: {
+      a: colors.menuItemText ?? 'rgba(0,0,0,0.6)',
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:35]: this code looks weird: 
you are setting one color and if it's undefined you are setting the default value,
but you have that colors everytime, they are static, and unnecessary conditions should be removed in all places

File: app/communities-configs/filecoin/index.js (hunk 5 above)
Comment (ruslanzharkov) [reply app/communities-configs/filecoin/index.js:54]: and check all places

File: app/communities-configs/filecoin/index.js [hunk 11]
```diff
@@ -0,0 +1,318 @@
...
+        font-size: smaller;
+        overflow-wrap: break-word;
+        white-space: normal;
+        margin-top: 10px;
+        line-height: 1.4;
+      }*/
+
+      @media only screen and (max-width: 991px) {
+        a {
+          padding: 0 0 0 30px  !important;
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:103]:
```suggestion
          padding: 0 0 0 30px !important;
```

File: app/communities-configs/filecoin/index.js [hunk 12]
```diff
@@ -0,0 +1,318 @@
...
+        }
+      }
+
+      @media only screen and (min-width: 992px) {
+        a {
+          padding: 5px 0 !important;
+          line-height: 1.6!important;
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:110]:
```suggestion
          line-height: 1.6 !important;
```

File: app/communities-configs/filecoin/index.js [hunk 13]
```diff
@@ -0,0 +1,318 @@
...
+      @media only screen and (min-width: 992px) {
+        padding: 0;
+        margin-left: 30px;
+
+        :first-child {
+          margin-left: 0;
+        }
+
+        :last-child {
+          margin-left: auto!important;
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:143]:
```suggestion
          margin-left: auto !important;
```

File: app/communities-configs/filecoin/index.js [hunk 14]
```diff
@@ -0,0 +1,318 @@
...
+          padding: 0!important;
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:144]:
```suggestion
          padding: 0 !important;
```

File: app/communities-configs/filecoin/index.js [hunk 15]
```diff
@@ -0,0 +1,318 @@
...
+        }
+      }
+
+      margin: 0 !important;
+    `,
+    CustomSubHeader: `
+      margin: 0 !important;
+    `,
+    subHeaderContainerStyles: `
+      // margin: 0 !important;
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:154]: remove this one

File: app/communities-configs/filecoin/index.js [hunk 16]
```diff
@@ -0,0 +1,318 @@
...
+    `,
+    topLogoContainerStyles: `
+      width: 80px!important;
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:157]: add empty spaces before `!important` construction

File: app/communities-configs/filecoin/index.js [hunk 17]
```diff
@@ -0,0 +1,318 @@
...
+  mobileSubHeader: (
+    <CustomMobileSubHeader config={CustomSubHeaderConfig} logo={FileiconLogo} />
+  ),
+  customSubHeader: <CustomSubHeader config={CustomSubHeaderConfig} />,
+  withoutFAQ: true,
+  fonts: {
+    h3LetterSpacing: '1px',
+    tagsLetterSpacing: '1px',
+    questionTitleLetterSpacing: '1px',
+    h3: 'Roboto-Regular, Arial, sans-serif',
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:267]: `Roboto-Regular, Arial, sans-serif` repeats for than 5 times for this file, extract to variable

File: app/communities-configs/filecoin/index.js [hunk 18]
```diff
@@ -0,0 +1,318 @@
...
+  },
+  fontFace: `@font-face {
+    font-family: 'Roboto-Regular';
+    src:
+      url(${RobotoRegularEOT}?#iefix) format('embedded-opentype'),
+      url(${RobotoRegularWOFF}) format('woff'),
+      url(${RobotoRegularTTF}) format('truetype');
+    font-style: normal;
+  }`,
+
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:300]: unnecessary empty line

File: app/communities-configs/filecoin/index.js [hunk 19]
```diff
@@ -0,0 +1,318 @@
...
+  elText: '#000000',
+  mainText: '#7A7A7A',
+  link: '#7A7AF0',
+  bg: '#FFFFFF',
+  extraBg: '#F3F3F3',
+  menuItemText: '#999999',
+  selectedMenuItemText: '#000000',
+};
+
+export const CustomSubHeaderConfig = {
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:27]: should be named in camelCase js format:
`customSubHeaderConfig`

File: app/components/CustomSubHeaders/CustomSubHeader.jsx [hunk 20]
```diff
@@ -154,7 +154,7 @@ const SubitemsTitle = styled.span`
     opacity: 0.6;
   }
 
-  transition: color 170ms ease-in-out;
+  transition: all 170ms ease-in-out;
```
Comment (ruslanzharkov) [inline app/components/CustomSubHeaders/CustomSubHeader.jsx:157]: if transition is the same, you can reuse it:
```
const transitionStyle = css`
  transition: all 170ms ease-in-out;
`;

const StyledDiv = styled.div`
  ${transitionStyle};
`
```

please do it here

File: app/components/CustomSubHeaders/CustomSubHeaderContainers/FilecoinStyleSubHeader.jsx [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+          subHeaderLogoStyles={styles.subHeaderLogoStyles}
+        >
+          <img src={src} alt="logo" />
+          {logoTitleText && (
+            <LogoTitle styles={logoTitleStyles}>{logoTitleText}</LogoTitle>
+          )}
+        </SubHeaderLogo>
+      </TopLogoContainer>
+      <TopMenuContainer>
+        {!!links && !!styles ? (
```
Comment (ruslanzharkov) [inline app/components/CustomSubHeaders/CustomSubHeaderContainers/FilecoinStyleSubHeader.jsx:84]: `!!(links && styles) ?`

File: server/port.js [hunk 22]
```diff
@@ -1,3 +1,3 @@
 const argv = require('./argv');
 
-module.exports = parseInt(argv.port || process.env.PORT || '3000', 10);
+module.exports = parseInt(argv.port || process.env.PORT || '31000', 10);
```
Comment (ruslanzharkov) [inline server/port.js:3]: rollback

**Review by ruslanzharkov — APPROVED** [review]

File: app/communities-configs/filecoin/index.js (hunk 5 above)
Comment (IgorAlexeew) [reply app/communities-configs/filecoin/index.js:54]: > please use `css` from styled-components

I didn't quite understand what was meant. This properties will be inserted as strings in styled-components in common components. I took this pattern from community config for another community.

File: app/communities-configs/filecoin/index.js [hunk 23]
```diff
@@ -0,0 +1,263 @@
...
+    }
+  `,
+  boostWalletBtnStyles: css`
+    g {
+      fill: #000000;
+    }
+  `,
+  headerHeight: 170,
+  isDropdownMenuArrow: false,
+  customSubHeaderConfig: customSubHeaderConfig,
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:261]:
```suggestion
  customSubHeaderConfig,
```

File: app/communities-configs/filecoin/index.js [hunk 24]
```diff
@@ -0,0 +1,263 @@
...
+  domainName: domainName,
```
Comment (ruslanzharkov) [inline app/communities-configs/filecoin/index.js:262]:
```suggestion
  domainName,
```

**Review by ruslanzharkov — APPROVED** [review]

**Pull request merged**